templates/graph.html

```diff
@@ -4,105 +4,176 @@
+                range: [-15, 10],
                 title: 'Force (N)'
             },
             title: 'Force (X-Direction) vs. Time Graph'
         };
 
         var layout2 = {
             xaxis: {
-                range: [0, 0.005],
+                range: [0, 3],
                 title: 'Time (s.)'
             },
             yaxis: {
-                range: [-0.25, 0.25],
+                range: [-25, 25],
                 title: 'Force (N)'
             },
-            title: 'Force (Y-Direction) vs. Time Graph'  // Different title for the second plot
+            title: 'Force (Y-Direction) vs. Time Graph'
         };
 
         Plotly.newPlot('plot', [trace1], layout1);
-        Plotly.newPlot('plot2', [trace2], layout2);  // Render the second plot
-
+        Plotly.newPlot('plot2', [trace2], layout2);
 
-
-          // Function used to display value of Impulse for X-Direction
-          function updateValueX(value) {
+        function updateValueX(value) {
             document.getElementById('value-container').innerHTML = '<b>Impulse (X-Direction):</b> ' + value + ' J';
         }
 
-        // Function used to display value of Impulse for Y-Direction
         function updateValueY(value) {
             document.getElementById('value-container').innerHTML = '<b>Impulse (Y-Direction):</b> ' + value + ' J';
         }
 
-        // Must click on datapoint you want integration to go up to for the first plot
         document.getElementById('plot').on('plotly_click', function(data){
             var pointIndex = data.points[0].pointIndex;
             var value = timex[pointIndex]; 
             updateValueX(value);
         });
 
-        // Must click on datapoint you want integration to go up to for the second plot
         document.getElementById('plot2').on('plotly_click', function(data){
             var pointIndex = data.points[0].pointIndex;
             var value = timey[pointIndex]; 
             updateValueY(value);
         });
 
-        // Function to navigate back to the landing page
         function goBack() {
             window.location.href = '/';
         }
+
+        function animatePlot(videoTime) { //videotime describes the time in the video that is input into the program
+        var frameIndex = findNearestFrameIndex(timex, videoTime); // This line does a search (like binary or cheetah) looking for a index that matches the videoTime varaible 
+        var frames = [{
+        data: [{x: timex.slice(0, frameIndex+1), y: forcex.slice(0, frameIndex+1)}] //Creates array with frameIndex
+        }];
+    
+        Plotly.animate('plot', frames, {
+        transition: {
+            duration: 0 //transitions right away
+        },
+        frame: {
+            duration: 0, // Instantly change to the new frame without any animation effect
+            redraw: false
+        }
+        });
+    }
+
+    function animatePlot(videoTime) { //videotime describes the time in the video that is input into the program
```

Comment:
duplicate function?

Reply:
"plot" is the x-direction force and "plot2" is the y-direction force. I will combine them and orientate them better with the video in a later commit.

Reply:
Oh I see did you intend to give them different names? I noticed in lines 171 you call the function `animatePlot2`

Reply:
Yes, those two different animate plot and animateplot2 are for the two different graphs.

Reply:
Right now they have the same name, both functions are called `animatePlot`